WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -19,30 +19,173 @@ final class ReviewsDashboardCardViewModel: ObservableObject {
+            stores.dispatch(ProductReviewAction.synchronizeProductReviews(siteID: siteID,
+                                                                          pageNumber: 1,
+                                                                          pageSize: Constants.numberOfItems) { result in
+                continuation.resume(with: result)
+            })
+        }
+    }
+
+    @MainActor
+    func retrieveProducts(for reviewProductIDs: [Int64]) async throws -> (products: [Product], hasNextPage: Bool) {
+        try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(ProductAction.retrieveProducts(siteID: siteID,
+                                                           productIDs: reviewProductIDs
+                                                          ) { result in
+                continuation.resume(with: result)
+            })
+        }
+    }
+
+    /// Performs initial fetch from storage and updates results.
+    func configureResultsControllers() {
+        configureProductReviewsResultsController()
+        configureProductsResultsController()
+    }
+
+    func configureProductReviewsResultsController() {
+        productReviewsResultsController.onDidChangeContent = { [weak self] in
+            guard let self else { return }
+            Task {
+                await self.updateReviewsResults()
+            }
+        }
+        productReviewsResultsController.onDidResetContent = { [weak self] in
+            guard let self else { return }
+            Task {
+                await self.updateReviewsResults()
+            }
+        }
+
+        do {
+            try productReviewsResultsController.performFetch()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
+    }
+
+    func configureProductsResultsController() {
+        productsResultsController.onDidChangeContent = { [weak self] in
+            self?.updateProductsResults()
+        }
+        productsResultsController.onDidResetContent = { [weak self] in
+            self?.updateProductsResults()
+        }
+
+        do {
+            try productsResultsController.performFetch()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
+    }
+
+    /// Updates data
+    @MainActor
+    func updateReviewsResults() async {
+        let reviews = productReviewsResultsController.fetchedObjects.prefix(3)
+        let reviewProductIDs = reviews.map { $0.productID }
+
+        // Load products that matches the review product IDs
+        if reviewProductIDs.isNotEmpty {
+            await loadReviewProducts(for: reviewProductIDs)
+        }
+
+        data = reviews
+            .map { review in
+                let product = reviewProducts.first { $0.productID == review.productID }
+                // TODO: also fetch notification
+                return ReviewViewModel(review: review, product: product, notification: nil)
+            }
```

Comment:
I'd like to get more opinion on this area.

The fetch result for reviews only includes data for product ID related to the reviews, while the UI requires the product name to be displayed. Thus, another call for `loadReviewProducts()` is needed to get the product name. 

I'm not 100% sure if this is the best way to do this.